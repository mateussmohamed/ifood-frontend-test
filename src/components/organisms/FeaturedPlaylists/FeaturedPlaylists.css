.featured-playlists {
  padding: 0 3rem;
  display: flex;
  flex-direction: column;
}

.featured-playlists-message {
  width: 100%;
}

.featured-playlists-cards {
  width: 100%;
  display: flex;
  flex-direction: row;
  overflow-x: auto;
}
