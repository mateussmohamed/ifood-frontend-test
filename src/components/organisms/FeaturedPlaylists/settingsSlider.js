export default {
  dots: false,
  infinite: false,
  speed: 500,
  slidesToShow: 4,
  slidesToScroll: 4,
  initialSlide: 0,
  className: 'featured-playlists-cards',
  responsive: [{
    breakpoint: 1024,
    settings: {
      slidesToShow: 3,
      slidesToScroll: 3,
      infinite: true,
      dots: false,
    },
  }, {
    breakpoint: 768,
    settings: {
      slidesToShow: 2,
      slidesToScroll: 2,
      initialSlide: 2,
    },
  }, {
    breakpoint: 480,
    settings: {
      slidesToShow: 1,
      slidesToScroll: 1,
    },
  },
  {
    breakpoint: 320,
    settings: {
      slidesToShow: 1,
      slidesToScroll: 1,
    },
  }],
};
