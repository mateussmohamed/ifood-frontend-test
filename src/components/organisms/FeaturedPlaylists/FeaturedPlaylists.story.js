import { storiesOf } from '@storybook/react';
import React from 'react';
import FeaturedPlaylists from './index';

const stories = storiesOf('FeaturedPlaylists', module);

const data = {
  message: 'As mais tocadas',
  playlists: {
    items: [
      {
        name: 'Bailando',
        tracks: { total: 60 },
        images: [
          {
            height: 300,
            url: 'https://i.scdn.co/image/723b3060add0aff0011fd2ae7d59237bbe788bb2',
            width: 300,
          },
        ],
      },
      {
        name: 'Esquenta Sertanejo',
        tracks: { total: 50 },
        images: [
          {
            height: 300,
            url: 'https://i.scdn.co/image/ac3e0fa062dbd40285b288fb7b17625dcd1948e1',
            width: 300,
          },
        ],
      },
      {
        name: 'Funk 25',
        tracks: { total: 25 },
        images: [
          {
            height: 300,
            url: 'https://i.scdn.co/image/f45205a65ddcca6ce2eef1cbc93c84fbc2503e72',
            width: 300,
          },
        ],
      },
    ],
  },
};

stories.add('default', () => <FeaturedPlaylists data={data} action={e => () => alert(e)} />);
