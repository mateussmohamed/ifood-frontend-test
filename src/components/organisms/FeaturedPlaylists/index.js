import React from 'react';
import PropTypes from 'prop-types';

import Card from '../../molecules/Card';
import Message from '../../atoms/Message';

import './FeaturedPlaylists.css';

const renderFeaturedPlaylists = (data, action) => (
  <div className="featured-playlists-cards">
    {data.map(item => (
      <Card
        key={item.id}
        title={item.name}
        image={item.images[0].url}
        songs={item.tracks.total}
        action={action(item.tracks.href)}
      />
    ))}
  </div>
);

const FeaturedPlaylists = ({ data, action }) => {
  const { playlists = {}, error = {}, message } = data;
  const msg = error.message || message;
  const type = error.message ? 'error' : 'info';
  return (
    <section className="featured-playlists">
      <div className="featured-playlists-message">
        <Message text={msg} type={type} />
      </div>
      {playlists.items && renderFeaturedPlaylists(playlists.items, action)}
    </section>
  );
};

FeaturedPlaylists.defaultProps = {
  data: [],
};

FeaturedPlaylists.propTypes = {
  data: PropTypes.objectOf(PropTypes.any),
  action: PropTypes.func.isRequired,
};

export default FeaturedPlaylists;
