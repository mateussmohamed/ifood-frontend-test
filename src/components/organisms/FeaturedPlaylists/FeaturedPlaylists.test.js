import React from 'react';
import renderer from 'react-test-renderer';
import FeaturedPlaylists from './index';

it('Should FeaturedPlaylists match to snapshot', () => {
  const component = renderer.create(<FeaturedPlaylists data={{}} action={e => () => alert(e)} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
