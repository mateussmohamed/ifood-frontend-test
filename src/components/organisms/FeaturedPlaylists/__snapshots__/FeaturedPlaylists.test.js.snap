// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FeaturedPlaylists match to snapshot 1`] = `
<section
  className="featured-playlists"
>
  <div
    className="featured-playlists-message"
  >
    <div
      className="message info"
    >
      Select filter for show playlists.
    </div>
  </div>
</section>
`;
